resources/schemas/dbscripts/postgresql/cds-22.005-22.006.sql

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2022 LabKey Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+TRUNCATE TABLE cds.import_studyReport;
+DROP TABLE cds.import_studyReport;
+
+CREATE TABLE cds.import_studyReport
+(
+    rowId SERIAL,
+    prot VARCHAR(250) NOT NULL,
+    cds_report_id INTEGER,
+    cds_report_link TEXT,
+    cds_report_label TEXT,
+    container ENTITYID NOT NULL,
+
+    CONSTRAINT PK_CDS_IMPORT_STUDY_REPORT PRIMARY KEY (rowId),
+    CONSTRAINT UQ_CDS_IMPORT_STUDY_REPORT UNIQUE (prot, cds_report_id, container, cds_report_link, cds_report_label)
+);
+
+TRUNCATE TABLE cds.studyReport;
+DROP TABLE cds.studyReport;
+
+CREATE TABLE cds.studyReport
+(
+    rowId SERIAL,
+    prot VARCHAR(250) NOT NULL,
+    cds_report_id INTEGER,
+    cds_report_link TEXT,
+    cds_report_label TEXT,
+    container ENTITYID NOT NULL,
+
+    CONSTRAINT PK_CDS_STUDY_REPORT PRIMARY KEY (rowId),
+    CONSTRAINT UQ_CDS_STUDY_REPORT UNIQUE (prot, cds_report_id, container, cds_report_link, cds_report_label)
+);
```

Comment:
Won't this force them to re-import all of their data after this upgrade? In the past wouldn't we just add new fields with reasonable defaults but preserve existing rows. It makes it a little trickier because you have to drop and then re-add constraints, see `cds-22.001-22.002.sql` for example.

Reply:
Correct it would force them to re-import, which they do anyway, dont' they normally?
I had implemented like you suggested previously, but ended up going this route because for some reason cds_report_id was showing up as a Required field even after dropping the PK constraint it was part of.

Reply:
From my local history, this is what i had done previously to drop the PK constraint, where cds_report_id was part of the PK ( 'PRIMARY KEY (prot, cds_report_id, container)')

ALTER TABLE cds.import_studyReport DROP CONSTRAINT PK_ImportStudyReport; 
ALTER TABLE cds.studyReport DROP CONSTRAINT PK_studyReport;

And it dropped the constraints as expected, but cds_report_id kept showing up as a required field (i also tried truncating first)

Reply:
They only run the ETLs when they have new data that they want to expose in the app. Looking at their production server, it looks like the last time they ran that ETL was in March of this year.

That SQL looks correct, it would be good if we can get it to work without having to delete any rows but if that isn't possible we will need to instruct them to run that ETL on both deployments.

Reply:
Looks like column part of PK gets set as NOT NULL by the db, once i dropped NOT NULL constraint on that column it now works as expected. 
Thanks!